From 0e4e1f61d1196df7415ee143804debcfe0ee88dd Mon Sep 17 00:00:00 2001
From: Parinay Kondekar <parinay.kondekar@seagate.com>
Date: Mon, 20 Jun 2016 17:00:59 +0530
Subject: [PATCH] LU-8259 osd-ldiskfs update all *_seq_write() to use "const
 char __user *buffer"

All *_seq_write functions in lustre/osd-ldiskfs/osd_lproc.c declare
buffer as "const char *buffer" and should use "const char __user *buffer"
instead.

This patch takes care of this change.

Change-Id: Ida44679f721a37cf71f327a6f00f3f8857ba0376
Signed-off-by: Parinay Kondekar <parinay.kondekar@seagate.com>
Reviewed-on: http://review.whamcloud.com/20887
Tested-by: Jenkins
Reviewed-by: John L. Hammond <john.hammond@intel.com>
Reviewed-by: James Simmons <uja.ornl@yahoo.com>
Tested-by: Maloo <hpdd-maloo@intel.com>
Reviewed-by: Oleg Drokin <oleg.drokin@intel.com>
---
 lustre/osd-ldiskfs/osd_lproc.c | 29 ++++++++++++++++-------------
 1 file changed, 16 insertions(+), 13 deletions(-)

diff --git a/lustre/osd-ldiskfs/osd_lproc.c b/lustre/osd-ldiskfs/osd_lproc.c
index 7eb806d186..599d7e20bf 100644
--- a/lustre/osd-ldiskfs/osd_lproc.c
+++ b/lustre/osd-ldiskfs/osd_lproc.c
@@ -173,10 +173,11 @@ static int osd_brw_stats_seq_show(struct seq_file *seq, void *v)
         return 0;
 }
 
-static ssize_t osd_brw_stats_seq_write(struct file *file, const char *buf,
-                                       size_t len, loff_t *off)
+static ssize_t osd_brw_stats_seq_write(struct file *file,
+				       const char __user *buf,
+				       size_t len, loff_t *off)
 {
-        struct seq_file *seq = file->private_data;
+	struct seq_file *seq = file->private_data;
         struct osd_device *osd = seq->private;
         int i;
 
@@ -274,7 +275,7 @@ static int ldiskfs_osd_cache_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_cache_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_cache_seq_write(struct file *file, const char __user *buffer,
 			    size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -309,7 +310,7 @@ static int ldiskfs_osd_wcache_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_wcache_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_wcache_seq_write(struct file *file, const char __user *buffer,
 				size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -332,7 +333,7 @@ ldiskfs_osd_wcache_seq_write(struct file *file, const char *buffer,
 LPROC_SEQ_FOPS(ldiskfs_osd_wcache);
 
 static ssize_t
-lprocfs_osd_force_sync_seq_write(struct file *file, const char *buffer,
+lprocfs_osd_force_sync_seq_write(struct file *file, const char __user *buffer,
 					size_t count, loff_t *off)
 {
 	struct seq_file	  *m = file->private_data;
@@ -362,7 +363,7 @@ static int ldiskfs_osd_pdo_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_pdo_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_pdo_seq_write(struct file *file, const char __user *buffer,
 				size_t count, loff_t *off)
 {
 	int rc;
@@ -391,7 +392,7 @@ static int ldiskfs_osd_auto_scrub_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_auto_scrub_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_auto_scrub_seq_write(struct file *file, const char __user *buffer,
 					size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -426,7 +427,8 @@ static int ldiskfs_osd_full_scrub_ratio_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_full_scrub_ratio_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_full_scrub_ratio_seq_write(struct file *file,
+				       const char __user *buffer,
 				       size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -467,7 +469,7 @@ static int ldiskfs_osd_full_scrub_threshold_rate_seq_show(struct seq_file *m,
 
 static ssize_t
 ldiskfs_osd_full_scrub_threshold_rate_seq_write(struct file *file,
-						const char *buffer,
+						const char __user *buffer,
 						size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -501,7 +503,7 @@ ldiskfs_osd_track_declares_assert_seq_show(struct seq_file *m, void *data)
 
 static ssize_t
 ldiskfs_osd_track_declares_assert_seq_write(struct file *file,
-						const char *buffer,
+						const char __user *buffer,
 						size_t count, loff_t *off)
 {
 	__s64 track_declares_assert;
@@ -542,7 +544,7 @@ static int ldiskfs_osd_readcache_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_readcache_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_readcache_seq_write(struct file *file, const char __user *buffer,
 				size_t count, loff_t *off)
 {
 	struct seq_file *m = file->private_data;
@@ -581,7 +583,8 @@ static int ldiskfs_osd_index_in_idif_seq_show(struct seq_file *m, void *data)
 }
 
 static ssize_t
-ldiskfs_osd_index_in_idif_seq_write(struct file *file, const char *buffer,
+ldiskfs_osd_index_in_idif_seq_write(struct file *file,
+				    const char __user *buffer,
 				    size_t count, loff_t *off)
 {
 	struct lu_env env;
-- 
2.17.1

